src/components/RangeSlider/components/SingleThumb/SingleThumb.scss

```diff
@@ -67,6 +83,33 @@ $range-thumb-shadow-focus: 0 0 0 rem(1px) color('indigo');
     var(--progress-upper, transparent) var(--Polaris-RangeSlider-progress, 100%),
     var(--progress-upper, transparent) 100%;
 
+  @mixin rangeSliderThumbPosition($size) {
+    margin-top: calc(-1 * (#{$size} - #{$range-track-height}) / 2);
+  }
+
+  &:hover,
+  &:active,
+  &:focus {
+    @include range-thumb-selectors {
+      transition-property: width, height, margin-top;
+      transition-duration: duration();
+      transition-timing-function: easing();
+
+      width: var(--p-range-slider-thumb-size-active, $range-thumb-size);
+      height: var(--p-range-slider-thumb-size-active, $range-thumb-size);
+
+      @include rangeSliderThumbPosition(
+        var(--p-range-slider-thumb-size-active, $range-thumb-size)
+      );
+
+      .disabled & {
+        width: $range-thumb-size;
+        height: $range-thumb-size;
+        @include rangeSliderThumbPosition($range-thumb-size);
+      }
```

Comment:
Although this seems a little repetitive, I didn't want to use the `:not()` function since it doesn't have great cross browser support.